src/main/java/blackjack/domain/BlackjackGame.java

```diff
@@ -0,0 +1,97 @@
+package blackjack.domain;
+
+import blackjack.domain.card.Card;
+import blackjack.domain.card.Deck;
+import blackjack.domain.scoreboard.GameResult;
+import blackjack.domain.scoreboard.ScoreBoard;
+import blackjack.domain.scoreboard.UserGameResult;
+import blackjack.domain.scoreboard.WinOrLose;
+import blackjack.domain.user.Dealer;
+import blackjack.domain.user.User;
+import blackjack.domain.user.Users;
+
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.toMap;
+
+public class BlackjackGame {
+    private static final String NO_MORE_PLAYING_USER_ERROR_MSG = "플레이 가능한 유저가 없습니다.";
+    private final Deck deck = Deck.createDeck();
+    private final Users users;
+    private final Dealer dealer = new Dealer();
+
+    private BlackjackGame(Users users) {
+        this.users = users;
+    }
+
+    public static BlackjackGame createAndFirstDraw(Users users) {
+        BlackjackGame blackjackGame = new BlackjackGame(users);
+        blackjackGame.init();
+        return blackjackGame;
+    }
+
+    public Dealer getDealer() {
+        return dealer;
+    }
+
+    public int calculateDealerScore() {
+        return dealer.calculateScore();
+    }
+
+    private void init() {
+        dealer.firstDraw(deck.draw(), deck.draw());
+        users.toList().forEach(user -> user.firstDraw(deck.draw(), deck.draw()));
+    }
+
+    public Card openDealerFirstCard() {
+        return dealer.getFirstCard();
+    }
+
+    public User findFirstCanPlayUser() {
+        return users.toList().stream()
+                .filter(User::canContinueGame)
+                .findFirst().orElseThrow(() -> new IllegalArgumentException(NO_MORE_PLAYING_USER_ERROR_MSG));
+    }
+
+    public Card draw() {
+        return deck.draw();
+    }
+
+    public List<User> getUsers() {
+        return Collections.unmodifiableList(users.toList());
+    }
+
+    public List<String> getUserNames() {
+        return users.toList().stream()
+                .map(User::getName)
+                .collect(Collectors.toList());
+    }
+
+    public ScoreBoard createScoreBoard() {
+        return new ScoreBoard(
+                users.toList().stream()
+                        .collect(
+                                toMap(Function.identity(), this::createGameResult, (exist, newer) -> newer, LinkedHashMap::new)
+                        )
+                , createDealerGameResult());
```

Comment:
따로 변수로 빼주는게 가독성이 더 좋을 것 같아요 :) 
```suggestion
        Map<User, UserGameResult> userResults = users.toList().stream()
                .collect(toMap(
                        Function.identity(),
                        this::createGameResult,
                        (exist, newer) -> newer, LinkedHashMap::new));
        return new ScoreBoard(userResults, createDealerGameResult());
```

Reply:
딜러 및 유저들의 카드 출력과 승패 출력 로직이 합쳐져 있어서 더 복잡해진 것 같아요. 
두 로직을 분리해보면 어떨까요? 

```
딜러 카드: 3다이아몬드, 9클로버, 8다이아몬드 - 결과: 20
pobi카드: 2하트, 8스페이드, A클로버 - 결과: 21
jason카드: 7클로버, K스페이드 - 결과: 17

## 최종 승패
딜러: 1승 1패
pobi: 승 
jason: 패
```

Reply:
카드출력과 승패출력을 분리해봤습니다! 근데 거의 컨트롤러에서 읽기 편하게 분리 시킨 느낌이라... 프로덕션 코드에서 수정해볼 부분을 놓치고 있는 것 같은데 힌트 주실 수 있나요?!

Reply:
`Map<User, UserGameResult> userResult` 에서 UserGameResult 내부에 유저들의 카드 정보와 승패 정보 둘다 들어가있어서 해당 정보를 만드느라 복잡해 보여요.
`UserGameResult` 내부에 유저의 카드정보를 넣을 필요없이 Participant에서 들고있는 Hand 데이터를 출력시켜줘도 되지않을까요? 
`UserGameResult`는 승,패 정보만 갖고있어도 충분해보여요.